lib/spandex_phoenix.ex

```diff
@@ -260,11 +260,17 @@ defmodule SpandexPhoenix do
   end
 
   defp replace_path_param_with_name(path_params, path_component) do
-    decoded_component = URI.decode(path_component)
+    decoded_component = uri_decode(path_component)
 
     Enum.find_value(path_params, decoded_component, fn
       {param_name, ^decoded_component} -> ":#{param_name}"
       _ -> nil
     end)
   end
+
+  defp uri_decode(path_component) do
+    URI.decode(path_component)
+  rescue
+    _error -> "<malformed_uri>"
```

Comment:
Few things (and sorry if I'm being too blunt in some parts - I'm trying to make sure that I'm understanding this correctly):

1. Using the original request path - that's a good point.  I thought that is our setup we could always pick up the original path from the Nginx log, but that would not work for everyone (and would put too much requirements on our logs setup).  In the end I wanted to reduce the number of resources in DD, but for those are responsible two different parts of the `%Spandex.Trace`.  So currently the `url:` will consist of full path, and the `resource:` will replace malformed parts with `<malformed>`.  And I've added log, just to make those errors more visible.
2. The resource name - current implementation is still not ideal, but I think it might be the most practical.  At the begging I thought that all those not parable uri should have one single resource `/<malformed>`, but this turns out to be tricky. It would require introducing some additional complexity, and with that some overhead. Leaving it as it is will produce resources which are mix of parsable and not parsable parts; something [like `GET /multiple/<malformed_uri>`](https://github.com/surgeventures/spandex_phoenix/pull/4/files#diff-32563a9c54d7d2adf1cd06f95b70cce481abdfbd6d3f1e134171044010b1b046R169).  This should be good enough, and without any additional overhead.
3. The detached tracer - I think I'm already doing that.  The `customize_metadata` is being called in the tracer process, and if it fails for any reason, the test process will not receive the processed trace.  So the test will fail [on the `assert_receive`](https://github.com/surgeventures/spandex_phoenix/pull/4/files#diff-32563a9c54d7d2adf1cd06f95b70cce481abdfbd6d3f1e134171044010b1b046R161) macro. Something like:

```elixir
     Assertion failed, no matching message after 100ms
     The process mailbox is empty.
     code: assert_receive {:sent_trace, %Spandex.Trace{spans: [%Spandex.Span{error: error, http: http, name: "request", resource: "GET /multiple/<malformed_uri>"}]}}
     stacktrace:
       test/tracer_integration/plug/router_test.exs:161: (test)
```

This still does not guarantee that all traces will succeed, or that the tracer will never be detached, but is should cover the case we discovered.  I've spend some time trying to find out if there are any other places where I could write any tests for this change, but came out empty.

Reply:
Point 1: I mostly agree, minus error logging (see my code review comment).
Point 2: Fully agree.
Point 3: If you mean my comment about us not testing telemetry handlers - what you're testing is the Router integration, but we're using the Telemetry integration (via `SpandexPhoenix.Telemetry.install/1`) and that's not tested at all.